.github/workflows/end2end.yml

```
@@ -0,0 +1,62 @@
+name: End-to-end test of EasyBuild in different distros
+on: [push, pull_request]
+jobs:
+  build_publish:
+    name: End-to-end test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        container:
+          - centos-7.9
+          - centos-8.4
+          - fedora-35
+          - opensuse-15.4
+          - rockylinux-8.5
+          - ubuntu-20.04
+      fail-fast: false
+    container:
+      image: ghcr.io/easybuilders/${{ matrix.container }}
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+
+      - name: download and unpack easyblocks and easyconfigs repositories
+        run: |
+          cd $HOME
+          for pkg in easyblocks easyconfigs; do
+              curl -OL https://github.com/easybuilders/easybuild-${pkg}/archive/develop.tar.gz
+              tar xfz develop.tar.gz
+              rm -f develop.tar.gz
+          done
+
+      - name: End-to-end test of EasyBuild
+        shell: bash
+        run: |
+          export PATH=$PWD:$PATH
+          export PYTHONPATH=$PWD:$HOME/easybuild-easyblocks-develop:$HOME/easybuild-easyconfigs-develop
+
+          # initialize environment (for Lmod)
+          if [ -f /etc/profile.d/modules.sh ]; then
+              # for Rocky, CentOS 8, Fedora
+              echo ">>> sourcing /etc/profile.d/modules.sh"
+              . /etc/profile.d/modules.sh
+          else
+              echo ">>> sourcing /etc/profile.d/*lmod*.sh"
+              . /etc/profile.d/*lmod*.sh
+          fi
+
+          # tests are run with root privileges, so we need to tell EasyBuild that's OK...
+          export EASYBUILD_ALLOW_USE_AS_ROOT_AND_ACCEPT_CONSEQUENCES=1
+
+          cmds=(
+            "env | sort"
+            "eb --version"
+            "eb --show-system-info"
+            "eb --check-eb-deps"
+            "eb --show-config"
+            "eb bzip2-1.0.8.eb --trace --robot"
+          )
+          for cmd in "${cmds[@]}"; do
+              echo ">>> $cmd"
+              eval "$cmd"
+          done
```

Following the same idea of having atomic steps, I would limit this one to sanity checks. By the way, I think that we can also simplify this step by removing the array of commands and just execute them straight away. Is there any purpose for the array?

```suggestion
          env | sort
          eb --version
          eb --show-system-info
          eb --check-eb-deps
          eb --show-config
```

---

I was using the array mostly so we can print the command being executed right before it's run.

That was useful when debugging this workflow, since the `eb` commands were failing in various interesting ways because the container images were rather minimal at first.

I prefer keeping it like this, since it's a bit easier to check the output produced by each of these commands if you also know which `eb` command was run...

---

ok, having the commands printed will be handy indeed. I'm in the totally opposite mindset and I would put each of these commands in its own `run` :smile: